org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/ParameterizedGenericMethodBinding.java

```diff
@@ -344,6 +344,7 @@ else if (invocationSite instanceof ReferenceExpression)
 								((ReferenceExpression) invocationSite).registerInferenceContext(methodSubstitute, infCtx18); // keep context so we can finish later
 						}
 					}
+					methodSubstitute.inferredReturnType= invocationTypeInferred;
```

Comment:
these two variables actually mean slightly different things:
* `inferredReturnType`: was the return type determined by type inference?
* `invocationTypeInferred`: has invocation type inference completed?